node/test/go/dev-mode/test/lib/process-function.js

```diff
@@ -0,0 +1,298 @@
+  const payload = { raw: String(Buffer.from(result.Payload)) };
+  log.debug('invoke response payload %s', payload.raw);
+  if (result.FunctionError) {
+    if (expectedOutcome.startsWith('error')) return duration;
+    throw new Error(`Invocation of ${testConfig.name} errored: ${result.FunctionError}`);
+  }
+  return { duration, payload };
+};
+
+const deleteFunction = async (testConfig) => {
+  await awsRequest(Lambda, 'deleteFunction', {
+    FunctionName: testConfig.configuration.FunctionName,
+  });
+};
+
+const retrieveReports = async (testConfig) => {
+  const retrieveEvents = async (nextToken = undefined) => {
+    const result = await awsRequest(CloudWatchLogs, 'filterLogEvents', {
+      startTime: testConfig.invokeStartTime,
+      logGroupName: `/aws/lambda/${testConfig.configuration.FunctionName}`,
+      nextToken,
+    });
+    // AWS happens to respond with `nextToken` chains when there are no further events
+    // Therefore we use it only if we received some events with the request
+    if (result.nextToken && result.events.length) {
+      return [...result.events, ...(await retrieveEvents(result.nextToken))];
+    }
+    return result.events;
+  };
+  const retrieveAllEvents = async () => {
+    await wait(1000);
+    try {
+      return await retrieveEvents();
+    } catch (error) {
+      if (error.name === 'ResourceNotFoundException') {
+        log.info('log group not ready, wait and retry %s', testConfig.name);
+        return retrieveAllEvents();
+      }
+      throw error;
+    }
+  };
+
+  let invocationsData;
+  let processesData;
+  do {
+    const events = await retrieveAllEvents();
+
+    let startEventsCount = 0;
+    let reportEventsCount = 0;
+    for (const { message } of events) {
+      if (message.startsWith('START RequestId: ')) ++startEventsCount;
+      else if (message.startsWith('REPORT RequestId: ')) ++reportEventsCount;
+    }
+
+    processesData = [];
+    invocationsData = [];
+    if (startEventsCount < testConfig.invokeCount || reportEventsCount < testConfig.invokeCount) {
+      continue;
+    }
+
+    log.debug('Events for %s: %o', testConfig.name, events);
+    let currentInvocationData;
+    let currentProcessData;
+    const getCurrentInvocationData = () => {
+      if (!currentInvocationData) {
+        log.error(
+          'Reject due to unexpected CW events (for start time %d): %o',
+          testConfig.invokeStartTime,
+          events
+        );
+        throw new Error(`Failed to resolve invocation data for ${testConfig.name}`);
+      }
+      return currentInvocationData;
+    };
+    for (const { message } of events) {
+      if (message.startsWith('⚡ DEV-MODE: initialization')) {
+        processesData.push((currentProcessData = { extensionOverheadDurations: {}, logs: [] }));
+        continue;
+      }
+      if (message.startsWith('⚡ DEV-MODE: Overhead duration: External initialization')) {
+        currentProcessData.extensionOverheadDurations.internalInit = parseInt(
```

Comment:
It is not `internalInit` (which suggests it's about internal extension initialization)